tests/system/src/test/java/test/robot/javafx/scene/ToolBarTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.robot.javafx.scene;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Orientation;
+import javafx.scene.AccessibleAttribute;
+import javafx.scene.control.Label;
+import javafx.scene.control.ToolBar;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.robot.Robot;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.WindowEvent;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import test.util.Util;
+
+public class ToolBarTest {
+
+    private static final double TOOLBAR_SIZE = 300.0;
+    private static final double CHILDREN_SIZE_TOO_BIG = 250.0;
+    private static final double ORIGINAL_CHILDREN_SIZE = 100.0;
+
+    private static final String ORIGINAL_STYLE = "-fx-border-width: 1; -fx-border-color: deepskyblue";
+    private static final String MODIFIED_STYLE = "-fx-border-width: 1; -fx-border-color: red";
+
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static Robot robot;
```

Comment:
robot is not at all used. 
The question is why should this test be under test.robot.javafx.scene package?

Reply:
Assuming that it even needs to be a system test, then `test/javafx/scene/control` seems an OK place.

My question is whether this needs to be a system test at all? If you can make it a javafx.controls unit test then that would be best (almost all controls tests can be done that way).

Reply:
If you decide to write a unit test - it can be added as a test case in existing test file rt/modules/javafx.controls/src/test/java/test/javafx/scene/control/ToolbarTest.java

else, system test under location suggested by Kevin looks fine.

Reply:
I think the system test is nice to have but it could make sense to have a unit test.
- I will try to implement one.
- I will move the test to the test/javafx/scene/control package.

Reply:
Tests move to proposed test class:
rt/modules/javafx.controls/src/test/java/test/javafx/scene/control/ToolbarTest.java

Reply:
To answer your question, by "javafx.controls unit test", I meant a test under `modules/javafx.controls/src/test/java/...` (as opposed to `tests/system/...`), which is what your latest change does. Did you verify that the test fails without your fix and passes with your fix?